### PR TITLE
Adding support to write Metadata to FieldType type via custom Attributes in Conventions.Net

### PR DESCRIPTION
**Pull request by clangelov**
this one would allow us to create out custom validation rules 

the PR contains some changes in line with the original Conventions.Net library

### CODE REVIEW

File: src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs [hunk 1]
```diff
@@ -29,45 +29,28 @@ public ISchema DeriveSchema(GraphSchemaInfo schemaInfo)
...
                 .Where(t => t.IsInterfaceType && !t.IsInputType && t.PossibleTypes.Any())
                 .GroupBy(t => t.Name)
-                .Select(g => g.First());                
-            var myPossibleTypes = interfaces
+                .Select(g => g.First());
+            var possibleTypes = interfaces
                 .Where(t => !t.IsIgnored)
                 .SelectMany(t => t.PossibleTypes)
-                .Select(x => OmitNonRootTypes(x))
+                .Select(typeInfo => (typeInfo.IsIgnored || !typeInfo.IsNullable || typeInfo.Interfaces.Any(x => x.IsIgnored == true))
```
Comment (Miragecoder) [inline src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs:36]: Maybe we can rewrite this like so:
```
.Where(typeInfo => !typeInfo.IsIgnored 
   || typeInfo.IsNullable 
   || typeInfo.Interfaces.All(x => !x.IsIgnored)
)
```
I think that way we can avoid the ternary/null and the `Where(x => x != null)`-operations here. 😃

File: src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs [hunk 2]
```diff
@@ -120,12 +104,31 @@ private FieldType DeriveField(GraphFieldInfo fieldInfo)
...
                 DeprecationReason = fieldInfo.DeprecationReason,
                 DefaultValue = fieldInfo.DefaultValue,
+                Metadata = DeriveFieldTypeMetaData(fieldInfo.AttributeProvider.GetCustomAttributes(false)),
                 Type = GetType(fieldInfo.Type),
                 Arguments = new QueryArguments(fieldInfo.Arguments.Where(arg => !arg.IsInjected).Select(DeriveArgument)),
                 Resolver = FieldResolverFactory(fieldInfo),
             };
         }
 
+        private IDictionary<string, object> DeriveFieldTypeMetaData(object[] attributes)
```
Comment (Miragecoder) [inline src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs:114]: I think we could also write this one like this:

```
if (attributes == null)
   return default;

return attributes
   .OfType<FieldTypeMetaDataAttribute>()
   .ToDictionary(x => x.Key(), x.Value());
```

Maybe we could consider in-lining it in the `DeriveType` that way as well.

**Review by Miragecoder — COMMENTED** [review]
I'm OK with these; some minor improvements are proposed. 😃

File: src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs (hunk 1 above)
Comment (clangelov) [reply src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs:36]: this is a result of merging with the latest Conventions.NET master branch, 
so I would avoid any changes here, I don't want to drag more things to support and maintain

File: src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs (hunk 2 above)
Comment (clangelov) [reply src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs:114]: Very good remark

**Pull request merged**